rasa/core/actions/action.py

```diff
@@ -116,22 +131,41 @@ def combine_with_templates(
     return actions + unique_template_names
 
 
+def is_retrieval_action(action_name: Text, retrieval_intents: List[Text]) -> bool:
+    """Check if an action name is a retrieval action.
+
+    The name for a retrieval action has an extra `utter_` prefix added to
+    the corresponding retrieval intent name.
+    Args:
+        action_name: Name of the action.
+        retrieval_intents: List of retrieval intents defined in the NLU training data.
+
+    Returns: True or False depending on whether the resolved intent name
+     is present in the list of retrieval intents.
+    """
+
+    return (
+        ActionRetrieveResponse.intent_name_from_action(action_name) in retrieval_intents
+    )
+
+
 def action_from_name(
     name: Text,
     action_endpoint: Optional[EndpointConfig],
     user_actions: List[Text],
     should_use_form_action: bool = False,
+    retrieval_intents: List[Text] = [],
```

Comment:
please NEVER use `[]` or `{}` in an argument list, they are mutable!
```suggestion
    retrieval_intents: Optional[List[Text]] = None,
```